src/containers/DownloadContainer/DownloadContainer.js

```diff
@@ -34,30 +34,33 @@ const DownloadContainer = ({ navigation }) => {
     }
   }
 
-  const fetchLocalFiles = async () => {
-    let files = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory)
-    files = files.filter((fileName) => fileName.endsWith(videoSuffix))
-    const videoItems = await Promise.all(
-      files.map(async (fileName) => {
-        const localUri = FileSystem.documentDirectory + fileName
-        const thumbnail = await generateThumbnail(localUri)
-        const filesize = await getFileSize(localUri)
-        const videoName = fileName.slice(0, -videoSuffix.length) // filename without suffix
-        return {
-          key: videoName,
-          name: videoName,
-          thumbnailImg: thumbnail,
-          filesize,
-          video: {
-            video1Path: fileName,
-          },
-        }
-      })
-    )
-    setLocalVideos(videoItems)
-  }
-
   useEffect(() => {
+    const fetchLocalFiles = async () => {
+      let files = await FileSystem.readDirectoryAsync(FileSystem.documentDirectory)
+      files = files.filter((fileName) => fileName.endsWith(videoSuffix))
+      const videoItems = await Promise.all(
+        files.map(async (fileName) => {
+          const localUri = FileSystem.documentDirectory + fileName
+          const thumbnail = await generateThumbnail(localUri)
+          const filesize = await getFileSize(localUri)
+          const videoName = fileName.slice(0, -videoSuffix.length) // filename without suffix
+          return {
+            key: videoName,
+            name: videoName,
+            thumbnailImg: thumbnail,
+            filesize,
+            video: {
+              video1Path: fileName,
+            },
+          }
+        })
+      )
+      // only update videoItems when videos changes
+      if (JSON.stringify(localVideos) !== JSON.stringify(videoItems)) {
```

Comment:
I still have the refreshing issue. There are two things causing this issue:
1. The thumbnailImg is always different since it is randomly generated
2. localVideos is always empty. When you use setInterval within a useEffect, any outside varaibles are captured at the time the interval is set. So changes to localVideos is not seen within the interval function, so this will always refresh.

I wrote some code to fix it, so I'll make another PR right now and request to merge it into this PR.

Reply:
Thanks. When I tested the codes on the iOS device, the refreshing problem was fixed. The Android platform might have a different mechanism. I would close this PR.